packages/ruleset/src/functions/parameter-case-convention.js

```diff
@@ -0,0 +1,84 @@
+const { casing } = require('@stoplight/spectral-functions');
+const { isSdkExcluded, isDeprecated } = require('../utils');
+
+// Error message prefix for each parameter type.
+const errorMsgPrefix = {
+  query: 'Query parameter names ',
+  path: 'Path parameter names ',
+  header: 'Header parameter names '
+};
+
+const errorMsgNoName = 'Parameters must have a name';
+const errorMsgNoIn = "Parameters must have a valid 'in' value";
+
+module.exports = function(param, options, { path }) {
+  return parameterCaseConvention(param, path, options);
+};
+
+/**
+ * This function will check 'param' to make sure that its name
+ * follows the appropriate case convention, depending on its parameter type.
+ * @param {*} param the parameter object to check
+ * @param {*} path the location of 'param' within the OpenAPI document
+ * @param {*} casingConfig this is the value of the 'functionOptions' field
+ * within this rule's definition (see src/rules/parameter-case-convention.js).
+ * This should be an object with an entry (key) for each parameter type (i.e. 'in' value).
+ * The value of each entry should be an object which is the config to be
+ * passed to Spectral's casing() function to enforce case conventions for that parameter type.
+ * @returns an array containing zero or more error objects
+ */
+function parameterCaseConvention(param, path, casingConfig) {
+  const errors = [];
```

Comment:
Like you've done for other rules, might be a good idea to defer this array creation until you've check for exclusion, etc. Probably doesn't make a big difference for this one since most parameters won't be excluded but thought I would mention it

Reply:
done